### PR TITLE
[MM-37064] - Onboarding flow is showing for existing users (#17977)

### PR DESCRIPTION
**Pull request by AGMETEOR**
Cherry-pick of #17977 to cloud

Release Note
```release-note
NONE

```

cc: @amyblais

### CODE REVIEW

File: model/preference.go [hunk 1]
```diff
@@ -13,23 +13,24 @@ import (
...
+	PreferenceCategorySidebarSettings   = "sidebar_settings"
+
+	PreferenceCategoryDisplaySettings     = "display_settings"
+	PreferenceNameCollapsedThreadsEnabled = "collapsed_reply_threads"
+	PreferenceNameChannelDisplayMode      = "channel_display_mode"
+	PreferenceNameCollapseSetting         = "collapse_previews"
+	PreferenceNameMessageDisplay          = "message_display"
+	PreferenceNameNameFormat              = "name_format"
+	PreferenceNameUseMilitaryTime         = "use_military_time"
+	PreferenceRecommendedNextSteps        = "recommended_next_steps"
```
Comment (marianunez) [inline model/preference.go:31]: I thought we had removed these extra ones from the original PR?

Comment (marianunez) [reply model/preference.go:31]: Oh, in `cloud` is not using camel case ? That's why this is a conflict?

Comment (AGMETEOR) [reply model/preference.go:31]: Happens when I try to cherry-pick and resolve conflicts

Comment (marianunez) [reply model/preference.go:31]: Ok, it was part of a 6.0 change to not use camel case anymore https://github.com/mattermost/mattermost-server/pull/17896

To be able to merge in cloud, we need to use the previous convention and avoid this conflict. You should be able to make the change in this `cherry-pick-MM-37064-cloud` branch

**Comment by AGMETEOR** [comment]
Closing in favor of #18002

**Pull request closed without merging**